core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -380,17 +382,33 @@ func newLegacyTransaction(nonce uint64, to common.Address, value *big.Int, gasLi
 	}
 }
 
-func signTx(keyStore KeyStore, address common.Address, tx *gethTypes.Transaction, chainID *big.Int) (common.Hash, []byte, error) {
+func SignTx(keyStore KeyStore, address common.Address, tx *gethTypes.Transaction, chainID *big.Int) (common.Hash, []byte, error) {
 	signedTx, err := keyStore.SignTx(address, tx, chainID)
 	if err != nil {
-		return common.Hash{}, nil, errors.Wrap(err, "signTx failed")
+		return common.Hash{}, nil, errors.Wrap(err, "SignTx failed")
 	}
 	rlp := new(bytes.Buffer)
-	if err := signedTx.EncodeRLP(rlp); err != nil {
-		return common.Hash{}, nil, errors.Wrap(err, "signTx failed")
+	if err = signedTx.EncodeRLP(rlp); err != nil {
+		return common.Hash{}, nil, errors.Wrap(err, "SignTx failed")
 	}
-	return signedTx.Hash(), rlp.Bytes(), nil
+	var hash common.Hash
+	hash, err = signedTxHash(signedTx, chainID)
+	if err != nil {
+		return hash, nil, err
+	}
+	return hash, rlp.Bytes(), nil
+}
 
+func signedTxHash(signedTx *gethTypes.Transaction, chainID *big.Int) (hash common.Hash, err error) {
+	if evmtypes.IsExChain(chainID) {
```

Comment:
Can we inject a `TxHasher` instead? The less refactoring we need to do for CC, the better

Reply:
When multichain is merged, I intend to refactor this to put it as a method on the chain object.

Until then, it makes it easier to refactor to do it like this.

Reply:
See: https://app.clubhouse.io/chainlinklabs/story/15991/refactor-okex-special-signing-logic-using-a-txhasher-or-similar